pytools/__init__.py

```diff
@@ -3014,13 +3013,22 @@ def opt_frozen_dataclass(
 
     .. versionadded:: 2024.1.18
     """
+
+    if "frozen" in kwargs:
+        raise ValueError("frozen must not be specified in opt_frozen_dataclass")
+
     def map_cls(cls: type[T]) -> type[T]:
         # This ensures that the resulting dataclass is hashable with and without
         # __debug__, unless the user overrides unsafe_hash or provides their own
         # __hash__ method.
+
+        # Make it possible to override 'frozen' in the class definition for testing.
+        # It would be nice to have something like https://discuss.python.org/t/allow-debug-to-be-set-at-runtime/64840
+        loc_frozen = __debug__ if "_frozen_override" not in cls.__dict__ else False
```

Comment:
Could you explain the use case?

Reply:
It's just meant for testing so that we can "fake" setting `__debug__=False`.